### PR TITLE
Update general understanding documents

### LINKED ISSUE #3585: correct WCAG 2.1 -> WCAG 2.2
In https://www.w3.org/WAI/WCAG22/Understanding/refer-to-wcag

### PR DESCRIPTION
**Pull request by patrickhlauke**
* removes the woefully out-of-date (and likely never going to be maintained) "Change log" from the https://www.w3.org/WAI/WCAG22/Understanding/about page
* where possible, makes WCAG mentions "versionless" ("WCAG 2" instead of "WCAG 2.0" / "WCAG 2.1" / "WCAG 2.2")
* <s>where not possible (e.g. links to techniques document), bumps to 2.2</s> EDIT: made techniques document link versionless as well
* <s>change `success criterion`/`success criteria` to `Success Criterion`/`Success Criteria` (and correct minor instances where the latter was used as if it were singular - "is the Success Criteria" > "are the Success Criteria" or "is the Success Criterion")</s> EDIT: as a result of discussions that came after this PR was made, reverting this change, and changing any straggler references that did use uppercase `Success Criteria` to lowercase `success criteria`
* adds `lang` attributes to all documents
* updates some of the links (most prominent: replace the outdated https://www.w3.org/WAI/GL/WCAG20/TECHS-SUBMIT/ to instead point to the github repo)

Closes #3585

Note that this change is based on existing non-normative information under [Background on WCAG 2](https://www.w3.org/TR/WCAG22/#background-on-wcag-2) in the existing TR, which reads in part:
>Where this document refers to WCAG 2 it is intended to mean any and all versions of WCAG that start with 2.

### CODE REVIEW

File: understanding/conformance.html
Comment (mbgower) [inline understanding/conformance.html:39]:
```suggestion
         <p>This means that if there is no content to which a Success Criterion applies, the Success
```

File: understanding/conformance.html
Comment (mbgower) [inline understanding/conformance.html:40]:
```suggestion
            Criterion is satisfied.
```

**Review by mbgower — APPROVED** [review]

**Comment by mbgower** [comment]
This verges on being only editorial, since it is not meaningfully changing the guidance. However, I have marked it as substantive since it is a bit of a departure.

File: understanding/documenting-accessibility-support.html
Comment (WilcoFiers) [inline understanding/documenting-accessibility-support.html:8]: Since we're working on WCAG 3, I think it would be better to keep a number in.

```suggestion
		<p>The documentation of accessibility support for uses of a Web technology provides the information needed to determine whether it is possible to satisfy the WCAG 2.2 Success Criteria for a particular environment. </p>
```

**Review by WilcoFiers — DISMISSED** [review]

**Comment by patrickhlauke** [comment]
Updated the PR, and made some further tweaks/changes along the way - see the updated description

**Review by mbgower — APPROVED** [review]
These seem like consistent changes, which are supported by some pre-existing uses of "WCAG 2" as well as the existing line in the introduction that establishes the use of this term across version releases:
> Where this document refers to WCAG 2 it is intended to mean any and all versions of WCAG that start with 2.

 By keeping the "2" it also distinguishes the information from any version 3 documents going forward.

**Review by bruce-usab — APPROVED** [review]
Thanks!  Edits look straightforward.

File: understanding/conformance.html
Comment (dbjorge) [inline understanding/conformance.html:159]: Should the link at the end of this sentence be generalized?

**Comment by patrickhlauke** [comment]
@mbgower @dav-idc made the `Success Criteria` change (and caught a few stragglers)

File: understanding/conformance.html
Comment (patrickhlauke) [reply understanding/conformance.html]: Done

File: understanding/conformance.html
Comment (dav-idc) [inline understanding/conformance.html:588]: Change to 'success criteron' to match the previous singular form earlier in the sentence.

File: understanding/understanding-techniques.html
Comment (dav-idc) [inline understanding/understanding-techniques.html:131]: Is the common term for the WCAG GitHub repository the 'WCAG source repository'? This is fine if that is the common name, I just figured I'd double-check that this is a consistent term.

File: understanding/understanding-techniques.html
Comment (dav-idc) [inline understanding/understanding-techniques.html:43]:
```suggestion
			<p><em>Techniques are informative — that means they are not required. The basis for determining conformance to WCAG 2 are the success criteria from the WCAG 2 standard — not the techniques.</em></p>
```

Matching the styling for em-dash interruptions that I noted in the `understanding-act-rules.html` page text. Having space between the words and em-dashes is also the formatting used on the `conformance.html` page.

Not a big deal either way, though, this is mainly a consistency thing.

File: understanding/intro.html
Comment (dav-idc) [inline understanding/intro.html:105]:
```suggestion
						<p>descriptions of the intent of the success criterion, including benefits, and examples.</p>
```

Recommend making 'success criteria' singular in this instance, as it's referring to info on a single criterion.

**Review by dav-idc — CHANGES_REQUESTED** [review]
Hi @patrickhlauke, I've completed my thorough review!

I only looked at portions of the pages that had changes, so this review doesn't guarantee that unchanged portions of the pages are perfectly matching the format of changed portions.

I've identified a few types of small consistency details to clean up. If you'd prefer, I can go in and commit the changes. Otherwise I'm happy for you to commit.

After resolving these comments and suggestions, I would consider my review complete.

File: understanding/understanding-techniques.html
Comment (patrickhlauke) [reply understanding/understanding-techniques.html]: honestly, not sure if there's an "official" term for it

**Comment by patrickhlauke** [comment]
@dav-idc wow, thank you for the thorough review, and catching the more insidious bits of "the original used criterion, but it should be criteria" and vice-versa. (note for myself next time...instead of accepting all suggestions individually, batch them)

**Review by dav-idc — APPROVED** [review]
With the pieces I reviewed implemented, looks good to me now!

File: understanding/documenting-accessibility-support.html
Comment (dav-idc) [reply understanding/documenting-accessibility-support.html]: Hi @WilcoFiers I think this PR now successfully switches any generic references to be `WCAG 2` and a few very specific WCAG 2.2 references to be `WCAG 2.2`. Would you mind reviewing one final time to unblock the PR?

File: understanding/intro.html
Comment (sebazai) [inline understanding/intro.html:88]: I was recently reading this page and I noticed that the Overview page said 13, and this page said 12. Not sure which one is correct.

```suggestion
				<p>Under each principle there is a list of guidelines that address the principle. There are a total of 13 guidelines. A convenient list of just the guidelines can be found in the <a class="gl-ref" href="https://www.w3.org/TR/WCAG/#toc">WCAG 2 table of contents</a>. One of the key objectives of the guidelines is to ensure that content is directly accessible to as many people as possible, and capable of being re-presented in different forms to match different peoples' sensory, physical and cognitive abilities.</p>
```

Comment (rscano) [reply understanding/intro.html:88]: actually (since wcag 2.1), 13.

Comment (mbgower) [reply understanding/intro.html:88]: Does that "convenient list of just the guidelines" work for people? I see it if I go to the original 2.0,  https://www.w3.org/TR/2008/REC-WCAG20-20081211/#contents , but not via the one at line 88

File: understanding/understanding-template.html
Comment (mbgower) [inline understanding/understanding-template.html:11]: I have a mild concern with removing the initial cap on "Guideline" here, simply because it specifically refers to the dozen or so Guidelines, but could be misread to refer to the entire document.
There is no way anyone could misunderstand "success criterion", but I think the capital G here helps understand what is being referred to.
```suggestion
			<p class="instructions">This section contains the main explanatory content of the Understanding. It explains why the Guideline or success criterion exists and, at a high level, how to meet it.</p>
```

File: understanding/understanding-template.html
Comment (mbgower) [inline understanding/understanding-template.html:40]: As per my prior comment, I'm inclined to leave all the G's capitalized in Guideline, when used in this sense. Don't feel super strong about it though.

File: understanding/about.html
Comment (gundulaniemann) [inline understanding/about.html:74]: Why is the changelog section deleted?
It is important to be able to retrieve when which piece was changed.

Comment (patrickhlauke) [reply understanding/about.html:74]: Because this has not been maintained at all in the last 5 years. It only contains a single item from one small change in 2019. The actual change/commit log is still available on github. Not sure it's worth keeping the section around at all, even if we removed that solitary single list item and just kept the reference to github.

File: understanding/understanding-template.html
Comment (shawna-slh) [reply understanding/understanding-template.html:40]: OK by me

File: understanding/about.html
Comment (gundulaniemann) [reply understanding/about.html:74]: I think it is valuable to keep the reference, so interested people can find the change log.

Comment (patrickhlauke) [reply understanding/about.html:74]: so just changing it to

```html
<details>
    <summary>
        <h2>Change Log</h2>
    </summary>
    <p>For a detailed view of recent changes to the informative documents see the <a href="https://github.com/w3c/wcag/commits/master">github updates</a>.</p>
</details>
```

(and leaving the link further up in the page to the change log section back in) ?

Comment (patrickhlauke) [reply understanding/about.html]: I've done the above (reintroduced the section, but only to point to github and nothing more)

File: understanding/understanding-techniques.html
Comment (patrickhlauke) [reply understanding/understanding-techniques.html]: we could also use "WCAG repository on GitHub" or similar

Comment (dav-idc) [reply understanding/understanding-techniques.html]: My recommendation:
`WCAG GitHub repository`

The URL points to GitHub anyhow, so there's not much benefit to keeping the language generic. A change in URL would be needed regardless, if we ever change technology.

Also, "WCAG GitHub repository" is searchable, and less likely to be mistaken for some other WCAG source-type thing we might have.

Comment (dav-idc) [reply understanding/understanding-techniques.html]: @patrickhlauke I'm cool with your suggestion of "WCAG repository on GitHub". This may not need to be thought through too much, as long as it clarifies what we're referring to 👍

File: understanding/understanding-techniques.html
Comment (dav-idc) [inline understanding/understanding-techniques.html:131]:
```suggestion
				<p>The WCAG Working Group encourages people to submit new techniques so that they can be considered for inclusion in updates of the <a href="../Techniques/">Techniques for WCAG 2</a> document. Please submit techniques for consideration to the <a href="https://github.com/w3c/wcag/">WCAG repository on GitHub</a>.</p>
```
Changing 'WCAG source repository' to 'WCAG repository on GitHub'. CC @patrickhlauke

File: understanding/understanding-template.html
Comment (patrickhlauke) [inline understanding/understanding-template.html:40]:
```suggestion
				<p>This section references techniques that can be used to meet the Guideline or success criterion. There are sub-sections for sufficient techniques, advisory techniques, and failures. </p>
```

File: understanding/intro.html
Comment (patrickhlauke) [reply understanding/intro.html]: `#toc` links you to the table of contents, which in the more recent (2.1 and 2.2) designs is on the left-hand side (so you won't visually see yourself being taken/scrolled anywhere)

File: understanding/documenting-accessibility-support.html
Comment (mbgower) [reply understanding/documenting-accessibility-support.html]: I believe what he has flagged has been resolved. @WilcoFiers I'll be marking this review as resolved and merging this today, unless you respond.
If you believe something needs to be resolved after the merge, we will prioritize review of it.

**Pull request merged**
